### ISSUE TITLE
Stocks fails by not grabbing symbol

### ISSUE THREAD

**Issue by duraki** [alexanderepstein/Bash-Snippets#76]
**Issue Label:**
* [x] Bug

**Description:**
**Stocks** for some reason returns unvalid stock symbol.

``` 
$ stocks AAPL                                                                                                         
/usr/local/bin/stocks: line 63:  5119 Done                    echo $stockInfo                                                          
5120 Segmentation fault: 11  | python -c "import sys, json; print json.load(sys.stdin)['Realtime Global Securities Quote']['02. E
xchange Name']" > /dev/null 2>&1                                                                                                       
Not a valid stock symbol
```

Tested when using `AAPL` and `Tesla`.

```
$ stocks -u
Bash-Snippets is already the latest version
```

**OS and OS version:**
* [x] Mac (Sierra, latest update)
* [ ] Linux 32 Bit
* [ ] Linux 64 Bit
* [ ] Windows 32 Bit
* [ ] Windows 64 Bit

OS Version: 10.12.5 (16F73)

**Comment by alexanderepstein** [comment]
Can you run ```bash -x stocks AAPL```and post the output here?

**Comment by duraki** [comment]
```
$ bash -x stocks AAPL > bash_snippets-76.txt

+ currentVersion=1.13.0
+ configuredClient=
+ configuredPython=
+ getConfiguredPython
+ command -v python2
+ command -v python
+ configuredPython=python
+ getConfiguredClient
+ command -v curl
+ configuredClient=curl
+ checkInternet
+ httpGet google.com
+ getopts uvh opt
+ [[ AAPL == \u\p\d\a\t\e ]]
+ [[ AAPL == \h\e\l\p ]]
+ [[ 1 == \0 ]]
+ getTicker AAPL
++ httpGet 'http://d.yimg.com/autoc.finance.yahoo.com/autoc?query=AAPL++++++++&region=1&lang=en%22'
--- CUT
```

The output is rather large so I will leave this [gist](https://gist.githubusercontent.com/duraki/1508ef3ece01e38af06eb8195cdad889/raw/949a4f1e1c22586a906dfddfb13c0bb670616552/bash_snipp.out.txt) here.
 
Also, installed over Homebrew.

**Comment by alexanderepstein** [comment]
It seems you get caught in some kind of infinite loop while trying to grab the stock symbol just posting the beginning of my output.
``` bash
+ currentVersion=1.13.0
+ configuredClient=
+ configuredPython=
+ getConfiguredPython
+ command -v python2
+ configuredPython=python2
+ getConfiguredClient
+ command -v curl
+ configuredClient=curl
+ checkInternet
+ httpGet google.com
+ getopts uvh opt
+ [[ AAPL == \u\p\d\a\t\e ]]
+ [[ AAPL == \h\e\l\p ]]
+ [[ 1 == \0 ]]
+ getTicker AAPL
```

The line ``` python -c 'import sys, json; print json.load(sys.stdin)['\''ResultSet'\'']['\''Result'\''][0]['\''symbol'\'']'``` only runs once for me. I will continue to look into this.

**Comment by duraki** [comment]
Ah I see. I think it might be with Yahoo service. Seems like it grabs my current location, and resolve a different URL based on IP, therefore parsing is not properly configured for that location page.

This is only a speculation.

I'll try to dig inside the script myself too.

**Comment by akantak** [comment]
@alexanderepstein 
Same here, installed with homebrew. It happens because symbol variable [here](https://github.com/alexanderepstein/Bash-Snippets/blob/master/stocks/stocks#L209) is not set. To tell the truth no variable that is using python will be set.
It took me a long while to debug it, but now I know why it doesn't work. The trouble starts [here](https://github.com/alexanderepstein/Bash-Snippets/blob/master/stocks/stocks#L109), where the symbol variable is not set because of your added method called python [here](https://github.com/alexanderepstein/Bash-Snippets/blob/master/stocks/stocks#L37). It does not forward further the output from python2 command execution, that's why.

**Comment by alexanderepstein** [comment]
> It does not forward further the output from python2 command execution, that's why.

Why do you think that is?

> because of your added method called python here

This was added because of #38 I am not quite sure why this doesn't work, on my linux machine I get ```configuredPython=python2``` and the commands calling python work perfectly. I think I may have fixed this by adding a OS check before creating my python function and the same OS check when I go to run ```getConfiguredClient```.

I am curious if the ```taste``` tool works for you guys. It uses the same ```getConfiguredPython``` principle.

**Comment by alexanderepstein** [comment]
Interesting it seems like the same issue is probably occurring but no error is ever being thrown I will add the same check to that script as well.

**Comment by akantak** [comment]
@alexanderepstein and just tested fix for stocks you mentioned. It is working fine, thanks!

**Comment by alexanderepstein** [comment]
Awesome, thank you for helping me pinpoint the issue, will release a new version by the end of today!

**Comment by alexanderepstein** [comment]
Released in [Version 1.14.3](https://github.com/alexanderepstein/Bash-Snippets/releases/tag/v1.14.3) Will be added to homebrew once @chainsawbaby updates his repository.
